scale.lua

```diff
@@ -63,12 +66,26 @@ function Scale:GetScale()
     return scale.x, scale.y
 end
 
+function Scale:SetFromScale(x, y)
+    self.from.x = x
+    self.from.y = y
+end
+
+function Scale:GetFromScale()
+    local scale = self.from
+
+    return scale.x, scale.y
+end
+
 function Scale:OnUpdate(elapsed)
     local properties = self.group.properties
 
+	local fromx = self.from.x ~= nil and self.from.x or 1
+	local fromy = self.from.y ~= nil and self.from.y or 1	
+	
     self.progress = self.smoothing_func(self.time / self.duration).y
 
     local frame = self.group.parent
-    frame:SetWidth(properties.width + self.progress * (properties.width * self.scale.x))
-    frame:SetHeight(properties.height + self.progress * (properties.height * self.scale.y))
+    frame:SetWidth(properties.width *  (fromx + self.progress *  (self.scale.x - fromx)))
+    frame:SetHeight(properties.height * (fromy + self.progress *  (self.scale.y - fromy)))
```

Comment:
```suggestion
    frame:SetWidth(properties.width * (from_x + self.progress * (self.scale.x - from_x)))
    frame:SetHeight(properties.height * (from_y + self.progress * (self.scale.y - from_y)))
```

The docs sounds like it's similar to that of `SetFromAlpha` and here you actually do fallback to its initial width if either scalar options isn't set.

Reply:
its also made for inscale of element so if i need scale from any scale to normal and stay normal scale after animation

i can show you how it works in MPOWA.